lib/Console/Command/Parse.php

```diff
@@ -23,37 +23,92 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Mesamatrix\Parser\OglVersion;
+use \Mesamatrix\Parser\UrlCache;
+use \Mesamatrix\Parser\Hints;
 
 class Parse extends \Symfony\Component\Console\Command\Command
 {
-    protected function configure()
-    {
+    protected $urlCache;
+    protected $statuses;
+
+    protected function configure() {
         $this->setName('parse')
              ->setDescription('Parse data and generate XML')
         ;
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        //$gitLog = exec_git("log --pretty=format:%H --reverse " .
-        //    Mesamatrix::$config->getValue("git", "oldest_commit").".. -- ".Mesamatrix::$config->getValue("git", "gl3"), $log);
-        //$initialCommit = rtrim(fgets($log));
-        $initialCommit = "master";
-
-        $cat = new \Mesamatrix\Git\ProcessBuilder(array(
-          'show', $initialCommit.':'.\Mesamatrix::$config->getValue('git', 'gl3')
+    protected function execute(InputInterface $input, OutputInterface $output) {
+        $gitLog = new \Mesamatrix\Git\ProcessBuilder(array(
+            'log', '--pretty=format:%H|%at|%aN|%cN', '--reverse',
+            \Mesamatrix::$config->getValue('git', 'oldest_commit').'..', '--',
+            \Mesamatrix::$config->getValue('git', 'gl3', 'docs/GL3.txt')
         ));
-        $proc = $cat->getProcess();
-        $this->getHelper('process')->mustRun($output, $proc);
-
-        $hints = new \Mesamatrix\Parser\Hints();
-        $parser = new \Mesamatrix\Parser\OglParser($hints);
-        $matrix = $parser->parse_content($proc->getOutput());
+        $gitLogProc = $gitLog->getProcess();
+        $this->getHelper('process')->mustRun($output, $gitLogProc);
+
+        $commitLines = explode(PHP_EOL, $gitLogProc->getOutput());
+        $hints = new Hints();
+        $matrix = new \Mesamatrix\Parser\OglMatrix();
+        foreach ($commitLines as $commitLine) {
+            list($commitHash, $time, $author, $committer) = explode('|', $commitLine);
+            $commit = new \Mesamatrix\Git\Commit($commitHash, $time, $author, $committer);
+            \Mesamatrix::$logger->info('Parsing GL3.txt for commit '.$commit->getHash());
+            $cat = new \Mesamatrix\Git\ProcessBuilder(array(
+              'show', $commit->getHash().':'.\Mesamatrix::$config->getValue('git', 'gl3')
```

Comment:
We use `\Mesamatrix::$config->getValue('git', 'gl3')` once at the beginning and once for each commits, it may be smart to cache it into a variable, what do you think?